docs/content/intro/tutorial-preview-environments.md

```diff
@@ -0,0 +1,424 @@
+
+We first comparing it with a list of `protected` namespace names and then we're using 
+API to check if the namespace already exists, and if not, we create it. 
+
+> **With power of Kubernetes API we can easily orchestrate any aspects of our pipeline.**
+
+## Environment Dependencies
+
+In this step we'll automate adding external dependencies like PostgreSQL to our new environment.
+
+```js
+const k8sAppClient = kubeConfig.makeApiClient(kubernetes.Apps_v1Api);
+
+const deployDependencies = async environmentName => {
+  const postgresqlStatefulSet = await k8sAppClient.listNamespacedStatefulSet(
+    environmentName,
+    undefined,
+    undefined,
+    undefined,
+    undefined,
+    "app=postgresql",
+  );
+  if (postgresqlStatefulSet.body.items.length) {
+    console.log("postgresql already deployed");
+  } else {
+    const postgresql = new Job("postgresql", "lachlanevenson/k8s-helm:v2.12.3");
+    postgresql.storage.enabled = false;
+    postgresql.imageForcePull = true;
+    postgresql.tasks = [
+      `helm init --client-only && \
+      helm repo update && \
+      helm upgrade ${environmentName}-postgresql stable/postgresql \
+      --install --namespace=${environmentName} \
+      --set fullnameOverride=postgresql \
+      --set postgresqlDatabase=products \
+      --set resources.requests.cpu=50m \
+      --set resources.requests.memory=156Mi \
+      --set readinessProbe.initialDelaySeconds=60 \
+      --set livenessProbe.initialDelaySeconds=60;`,
+    ];
+    await postgresql.run();
+  }
+};
+
+const provisionEnvironment = async (environmentName, projects) => {
+  await deployDependencies(environmentName);
+};
+
+```
+
+Before deploying our PostgreSQL StatefulSet we check if one already exists. 
+To list StatefulSets in a Namespace we need to use `listNamespacedStatefulSet` 
+api endpoint that lives in `kubernetes.Apps_v1Api` set of APIs. 
+We create instance of it called `k8sAppClient` first.
+
+Once we verify our dependency needs to be installed we create `postgresql` Job that will 
+run `lachlanevenson/k8s-helm:v2.12.3` image as it's worker. 
+Lachlan Evenson has been building and publishing [docker images](https://hub.docker.com/r/lachlanevenson/k8s-helm/) with every version of Helm.
+
+After Job successfully completes we can verify our PostgreSQL is installed:
+
+```console
+$ helm list
+
+NAME            	STATUS  	CHART                	NAMESPACE
+bob-postgresql  	DEPLOYED	postgresql-3.16.     	bob
+```
+
+## Environment ConfigMap
+
+Environment ConfigMap is created in `brigade` namespace where we keep track of all projects we would like to be installed in our environment. One ConfigMap is created per environment.
+
+Sample of data structure:
+
+```yaml
+projects:
+  products:
+    tag: prod
+  orders:
+    tag: prod
+```
+
+Our ConfigMap will be labeled with `type: preview-environment-config` label. This label will be used as a selector by application's `brigade.js` script at the time of a release.
```

Comment:
Missing `a`: `...will be labeled with a "type: preview-environment-config" label.`

Reply:
Missing article: `...will be used as a selector by the application's brigade.js...`